infrastructures/kubernetes/src/test/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/NamespaceNameValidatorTest.java

```diff
@@ -24,24 +25,47 @@
 
   @Test
   public void testFailsOnNull() {
-    assertEquals(NULL_OR_EMPTY, validateInternal(null));
+    assertEquals(validateInternal(null), NULL_OR_EMPTY);
   }
 
   @Test
   public void testFailsOnEmpty() {
-    assertEquals(NULL_OR_EMPTY, validateInternal(""));
+    assertEquals(validateInternal(""), NULL_OR_EMPTY);
   }
 
   @Test
   public void testFailsOnTooLong() {
     assertEquals(
-        TOO_LONG,
-        validateInternal("0123456789012345678901234567890123456789012345678901234567890123"));
+        validateInternal("0123456789012345678901234567890123456789012345678901234567890123"),
+        TOO_LONG);
   }
 
   @Test(dataProvider = "invalidDnsNames")
   public void testFailsOnInvalidChars(String invalidName) {
-    assertEquals(INVALID, validateInternal(invalidName));
+    assertEquals(validateInternal(invalidName), INVALID);
+  }
+
+  @Test(dataProvider = "invalidUsernames")
+  public void normalizeTest(String raw, String expected) {
+    assertEquals(expected, normalize(raw));
+  }
+
+  @Test
+  public void normalizeLengthTest() {
+    assertEquals(
+        63,
+        normalize("looooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooong")
+            .length());
+  }
+
+  @DataProvider
+  public static Object[][] invalidUsernames() {
+    return new Object[][] {
+      new Object[] {"gmail@foo.bar", "gmail-foo-bar"},
+      new Object[] {"_fef_123-ah_*zz**", "fef-123-ah-zz"},
+      new Object[] {"a-b#-hello", "a-b-hello"},
+      new Object[] {"a---------b", "a-b"},
```

Comment:
`new Object[] {"--ab--", "ab"},`

I think ^ is gonna fail.

Reply:
added such case